hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3ABlockOutputStream.java

```diff
@@ -389,8 +397,10 @@ public void close() throws IOException {
         // PUT the final block
         if (hasBlock &&
             (block.hasData() || multiPartUpload.getPartsSubmitted() == 0)) {
-          //send last part
-          uploadCurrentBlock();
+          // send last part and set the value of isLastPart to true in case of
+          // CSE being enabled, since we are sure it is last part as parts
+          // are being uploaded serially in CSE.
+          uploadCurrentBlock(isCSEEnabled);
```

Comment:
I was thinking about this myself but thought shouldn't we keep the non-CSE behavior as it was, that is pre this patch, we weren't setting the lastPart=true for non-CSE multi-part uploads, and what would be the implications with multiple threads as well? But it does make sense to have it true since it would always be the last part uploaded... @steveloughran thoughts??

Reply:
good point. Looking at source there's no indication this is bad news if other uploads are in progress, and it would simplify code.

why not set it?